flink-autoscaler/src/main/java/org/apache/flink/autoscaler/event/AutoScalerEventHandler.java

```diff
@@ -63,6 +65,19 @@ void handleEvent(
             @Nullable String messageKey,
             @Nullable Duration interval);
 
+    /**
+     * Handle exception, and the exception event is warning type and don't deduplicate by default.
+     */
+    default void handleException(Context context, String reason, Throwable e) {
+        var message = e.getMessage();
+        if (message == null) {
+            var stream = new ByteArrayOutputStream();
+            e.printStackTrace(new PrintStream(stream));
+            message = stream.toString();
```

Comment:
Can be replaced with `ExceptionUtils.getStackTrace(e)` from apache commons  or ExceptionUtis.stringifyException from flink utils

Reply:
we should also set some reasonable (relatively small) limit on the string size here otherwise we might get an error when inserting the Kube event / db column

Reply:
Thanks @gyfora for the quick review and good suggestion!

> Can be replaced with ExceptionUtils.getStackTrace(e) from apache commons or ExceptionUtis.stringifyException from flink utils

Done~

> we should also set some reasonable (relatively small) limit on the string size here otherwise we might get an error when inserting the Kube event / db column

I saw the default value of `kubernetes.operator.exception.stacktrace.max.length` is 2048, so I use it here.

And I didn't extract an autoscaler option for it unless it's needed in the future.